metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/stellar/GetProfile.java

```diff
@@ -226,14 +262,23 @@ private ColumnBuilder getColumnBuilder(Map<String, Object> global) {
    * @param global The global configuration.
    */
   private RowKeyBuilder getRowKeyBuilder(Map<String, Object> global) {
-    /*
-     * WARNING: the row key builder is not currently configurable.  by invoking
-     * the default constructor below, this defaults to generating keys using a
-     * period duration of 15 minutes. this function will NOT be able to read
-     * profiles created by a profiler running with any other period duration, but
-     * 15 minutes.
-     */
-    return new SaltyRowKeyBuilder();
+
+    // how long is the profile period?
+    String configuredDuration = (String) global.getOrDefault(PROFILER_PERIOD, PROFILER_PERIOD_DEFAULT);
+    long duration = Long.parseLong(configuredDuration);
+    LOG.debug("profiler client: {}={}", PROFILER_PERIOD, duration);
+
+    // which units are used to define the profile period?
+    String configuredUnits = (String) global.getOrDefault(PROFILER_PERIOD_UNITS, PROFILER_PERIOD_UNITS_DEFAULT);
+    TimeUnit units = TimeUnit.valueOf(configuredUnits);
+    LOG.debug("profiler client: {}={}", PROFILER_PERIOD_UNITS, units);
+
+    // what is the salt divisor?
+    String configuredSaltDivisor = (String) global.getOrDefault(PROFILER_SALT_DIVISOR, PROFILER_SALT_DIVISOR_DEFAULT);
+    int saltDivisor = Integer.parseInt(configuredSaltDivisor);
+    LOG.debug("profiler client: {}={}", PROFILER_SALT_DIVISOR, saltDivisor);
+
+    return new SaltyRowKeyBuilder(saltDivisor, duration, units);
```

Comment:
Okay, let's open a new jira for this too (separate from METRON-530, because they are completely separable and this is higher priority), and get on with this PR.